src/Engine/ProtoCore/FFI/CLRObjectMarshaler.cs

```diff
@@ -430,7 +513,114 @@ private ICollection ToICollection(StackValue dsObject, ProtoCore.Runtime.Context
             }
 
             return arrList;
+        }
+
+        private IList ToICollection(CLRStackValue dsObject, System.Type arrayType, ProtoCore.MSILRuntimeCore runtimeCore)
+        {
+            if (arrayType.IsArray)
+            {
+                //  processing only for the primitive types
+                //  anything else will be dealt with as it was earlier
+                //
+                if (arrayType.UnderlyingSystemType == typeof(int[]))
+                {
+                    return UnMarshal<int>(dsObject, runtimeCore);
+                }
+                else if(arrayType.UnderlyingSystemType == typeof(double[]))
+                {
+                    return UnMarshal<double>(dsObject, runtimeCore);
+                }
+                else if (arrayType.UnderlyingSystemType == typeof(bool[]))
+                {
+                    return UnMarshal<bool>(dsObject, runtimeCore);
+                }
+            }
+
+            var dsArrayValues = dsObject.Value as IList<CLRStackValue>;
+
+            //  use arraylist instead of object[], this allows us to correctly capture 
```

Comment:
comment seems out of date?

Reply:
actually that still applies (maybe not be best way..but it still works)